File: src/test/java/net/tlalka/imager/utils/RxUtilsText.java
package net.tlalka.imager.utils;

import io.reactivex.Flowable;
import io.reactivex.schedulers.Schedulers;
import org.junit.Test;

import static net.tlalka.imager.utils.RxUtils.comp;

public class RxUtilsText {

    @Test
    public void rxCompSequential() throws Exception {
        Flowable.range(1, 10)
                .observeOn(Schedulers.computation())
                .map(i -> i * 2)
                .scan(comp((i1, i2) -> i2 = 0))
                .blockingSubscribe(System.out::println);
    }

    @Test
    public void rxCompParallel() {
        Flowable.range(1, 10)
                .parallel()
                .runOn(Schedulers.computation())
                .map(i -> i * 2)
                .sequential()
                .scan(comp((i1, i2) -> i2 = 0))
                .blockingSubscribe(System.out::println);
    }
}

File: src/test/kotlin/net/tlalka/imager/utils/RxUtilsText.kt
package net.tlalka.imager.utils

import io.reactivex.Flowable
import io.reactivex.schedulers.Schedulers
import org.junit.Test

class RxUtilsText {

    @Test
    fun rxCompSequential() {
        Flowable.range(1, 10)
                .observeOn(Schedulers.computation())
                .map { i -> i!! * 2 }
                .scan { _, _ -> 0 }
                .blockingSubscribe(::println)
    }

    @Test
    fun rxCompParallel() {
        Flowable.range(1, 10)
                .parallel()
                .runOn(Schedulers.computation())
                .map { i -> i!! * 2 }
                .sequential()
                .scan { _, _ -> 0 }
                .blockingSubscribe(::println)
    }
}
